pkg/v1/providers/tests/unit/windows_test.go

```diff
@@ -50,15 +51,158 @@ var _ = Describe("Windows Ytt Templating", func() {
 			"TKG_DEFAULT_BOM":             "tkg-bom-v1.4.0.yaml",
 			"KUBERNETES_RELEASE":          "v1.21.2---vmware.1-tkg.1",
 			"CORE_DNS_IP":                 "10.64.0.10",
-			"CLUSTER_CIDR":                "192.168.1.0/16",
-		})
+			"IS_WINDOWS_WORKLOAD_CLUSTER": "true",
+			"TKG_CLUSTER_ROLE":            "workload",
+			"CLUSTER_NAME":                "win-wl",
+		}
+	})
+	It("Support multiple Windows workload nodes", func() {
+		workloadCount := 10
+		values["WORKER_MACHINE_COUNT"] = workloadCount
 
+		rawClusterAPIYaml, err := renderYTTTemplate(paths, values)
+		Expect(err).NotTo(HaveOccurred())
+
+		unstructObjs, err := searchObjectByKind("MachineDeployment", rawClusterAPIYaml)
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, unstruct := range unstructObjs {
+			replicas, _, err := unstructured.NestedFloat64(unstruct.Object, "spec", "replicas")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(replicas).To(Equal(float64(workloadCount)))
+		}
+	})
+	It("Has cluster_plan=prod cluster annotation", func() {
+		values["CLUSTER_PLAN"] = "prod"
+
+		rawClusterAPIYaml, err := renderYTTTemplate(paths, values)
+		Expect(err).NotTo(HaveOccurred())
+		obj, err := searchObjectByName("win-wl", rawClusterAPIYaml) // cluster object
+		Expect(err).NotTo(HaveOccurred())
+
+		compare := reflect.DeepEqual(obj.GetAnnotations(), map[string]string{"osInfo": ",,", "tkg/plan": "prod"})
+		Expect(compare).To(BeTrue())
+	})
+	It("Has correct kubeadmConfigTemplate for CNI:None", func() {
+		values["CNI"] = "none"
+		rawClusterAPIYaml, err := renderYTTTemplate(paths, values)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(rawClusterAPIYaml).To(ContainSubstring("CNI: none"))
```

Comment:
Can we do better tests than grep please? I've already refactored these tests once.